packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -146,35 +167,88 @@ export class GCFBootstrapper {
     return config as Options;
   }
 
+  parseRequestHeaders(
+    request: express.Request
+  ): {name: string; id: string; signature: string; taskId: string} {
+    const name =
+      request.get('x-github-event') || request.get('X-GitHub-Event') || '';
+    const id =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    // TODO: add test to validate this signature is used on the initial
+    // webhook from GitHub:
+    const signature =
+      request.get('x-github-delivery') ||
+      request.get('X-GitHub-Delivery') ||
+      '';
+    const taskId =
+      request.get('X-CloudTasks-TaskName') ||
+      request.get('x-cloudtasks-taskname') ||
+      '';
+    return {name, id, signature, taskId};
+  }
+
+  parseTriggerType(name: string, taskId: string): TriggerType {
+    if (
+      !taskId &&
+      (name === 'schedule.repository' || name === 'pubsub.message')
+    ) {
+      return TriggerType.SCHEDULER;
+    } else if (!taskId && name) {
+      return TriggerType.GITHUB;
+    } else if (name) {
+      return TriggerType.TASK;
+    }
+    return TriggerType.UNKNOWN;
+  }
+
+  buildTriggerInfo(
+    triggerType: TriggerType,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestBody: {[key: string]: any}
+  ) {
+    let triggerInfo = {trigger_type: triggerType};
+
+    if (triggerType === TriggerType.GITHUB) {
+      const sourceRepo = requestBody['repository'] || {};
+      const repoName: string = sourceRepo['name'] || 'UNKNOWN';
+      const repoOwner = sourceRepo['owner'] || {};
+      const ownerName: string = repoOwner['login'] || 'UNKNOWN';
+      const ownerType: string = repoOwner['type'] || 'UNKNOWN';
+      const sender = requestBody['sender'] || {};
+      const senderLogin: string = sender['login'] || 'UNKNOWN';
+
+      const webhookProperties = {
+        trigger_source_repo: {
+          repo_name: repoName,
+          owner: ownerName,
+          owner_type: ownerType,
+        },
+        trigger_sender: senderLogin,
+      };
+
+      triggerInfo = {...webhookProperties, ...triggerInfo};
+    }
+
+    return {trigger: triggerInfo};
+  }
+
   gcf(
     appFn: ApplicationFunction
   ): (request: express.Request, response: express.Response) => Promise<void> {
     return async (request: express.Request, response: express.Response) => {
       // Otherwise let's listen handle the payload
       this.probot = this.probot || (await this.loadProbot(appFn));
+      const {name, id, signature, taskId} = this.parseRequestHeaders(request);
+      const triggerType: TriggerType = this.parseTriggerType(name, taskId);
+
+      if (triggerType !== TriggerType.TASK) {
+        // we don't want to log TASK triggers since we enqueue the task ourselves
```

Comment:
I *think* tasks can be requeued on failure, so we might want to see trigger info on the task types?

Without logging the task executions, it still won't be easy to easily find the real logs we are looking for.

Reply:
Ideally when we enqueue a task after getting the webhook from Github, we would also pass on the execution id of the function that received the webhook so we could have a mapping between them. But yeah, your point makes sense. For now I'll just log all triggers as-is.

Reply:
Looks like it's not possible to access the `execution_id` from the function itself: https://groups.google.com/g/firebase-talk/c/5x4MtKxSdNQ

Reply:
Can we associate the events via the [GitHub delivery GUID](https://developer.github.com/webhooks/event-payloads/#delivery-headers)?

Reply:
Yeah I was going to say we can create our own GUID but this works too